batchflow/utils.py

```diff
@@ -8,6 +11,12 @@
 from matplotlib import pyplot as plt
 from matplotlib import colors as mcolors
 
+import ipykernel
+import requests
+from notebook.notebookapp import list_running_servers
+from nbconvert import PythonExporter
+from pylint import epylint as lint
```

Comment:
This will required a lot of packages to be pre-installed for batchflow even if you don't use these functions.
Make them optional.

Reply:
I can either use `try/except` block on top of the module, or move those imports to the functions that use them. What is the preferred way of making modules optional?

Reply:
Top imports are much more explicit, thus better.
But local imports are also acceptable if they are mentioned at the top.

Reply:
I would also suggest to split `utils` into separate modules for plotting, notebooks, etc